plugins/woocommerce/tests/metrics/specs/editor.spec.js

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable playwright/no-conditional-in-test, playwright/expect-expect */
+
+/**
+ * WordPress dependencies
+ */
+import { test, Metrics } from '@wordpress/e2e-test-utils-playwright';
+
+/**
+ * Internal dependencies
+ */
+import { PerfUtils } from '../fixtures';
+import { median } from '../utils';
+
+// See https://github.com/WordPress/gutenberg/issues/51383#issuecomment-1613460429
+const BROWSER_IDLE_WAIT = 1000;
+
+const results = {};
+
+async function editPost( admin, page, postId ) {
+	const query = new URLSearchParams();
+	query.set( 'post', String( postId ) );
+	query.set( 'action', 'edit' );
+
+	await admin.visitAdminPage( 'post.php', query.toString() );
+	await setPreferences( page, 'core/edit-post', {
+		welcomeGuide: false,
+		fullscreenMode: false,
+	} );
+}
+
+async function setPreferences( page, context, preferences ) {
+	await page.waitForFunction( () => window?.wp?.data );
+
+	await page.evaluate(
+		async ( props ) => {
+			for ( const [ key, value ] of Object.entries(
+				props.preferences
+			) ) {
+				await window.wp.data
+					.dispatch( 'core/preferences' )
+					.set( props.context, key, value );
+			}
+		},
+		{ context, preferences }
+	);
+}
```

Comment:
I assume these haven't been released with the core package yet. In such case, we can still extend the original fixture by doing the following:

```js
import { Editor } from '@wordpress/e2e-test-utils-playwright';

class EditorExtended extends Editor {
	async editPost( postId ) {
		// ...
	}

	async setPreferences( context, preferences ) {
		// ...
	}
}

test.use( {
	editor: async ( { page }, use ) => {
		await use( new EditorExtended( { page } ) );
	},
} );
```

This way, we don't need to update the tests.

Reply:
They have been released but there's some rules I guess in the repo about "fixing dependencies". I didn't want to update the dependency as it's used elsewhere and I don't know the consequences. But yeah, good call about the extension, I didn't know about it :)